include/swift/SIL/SILFunction.h

```diff
@@ -1234,6 +1234,29 @@ class SILFunction
     return V && V->getAttrs().hasAttribute<AlwaysEmitIntoClientAttr>();
   }
 
+  /// Returns true if this function belongs to a declaration that
+  /// has `@_used` attribute.
+  bool markedAsUsed() const {
+    if (!hasLocation())
+      return false;
+
+    auto *V = getLocation().getAsASTNode<ValueDecl>();
```

Comment:
It would be better to store the used and section info directly in SILFunction. Getting it from the decl via debug info does not survive serialization.

Reply:
Will switch to that. For my information, why isn't this a problem in the function above, `markedAsAlwaysEmitIntoClient`?

Reply:
In most cases it will work, because that info comes from the serialized AST (and not from SIL).
I'm not sure if this is really a problem in practice. It might be a problem for functions, which are created by the optimizer, like specializations, etc.